### PR TITLE
Change the APE field validation to match all formats

### PR DESCRIPTION
**Pull request by alexandrebak42**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop / 8.1.x
| Description?      | Change Validate::isApe method and PREFEX_customer table to match all APE or NAF formats.
| Type?             |  improvement 
| Category?         |  CO, IN 
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Change APE code to formats XX.XXX and XXXXX
| Fixed issue or discussion?     | Fixes #34216
| Sponsor company | coquille.fr

### CODE REVIEW

**Review by mflasquin — DISMISSED** [review]
LGTM, if this PR is merged, do not forget to add a PR to take changes into account in upgrade module https://github.com/PrestaShop/autoupgrade

**Review by matthieu-rolland — DISMISSED** [review]

File: classes/Validate.php [hunk 1]
```diff
@@ -1269,7 +1269,7 @@ public static function isSiret($siret)
      */
     public static function isApe($ape)
     {
-        return (bool) preg_match('/^[0-9]{3,4}[a-zA-Z]{1}$/s', $ape);
+        return (bool) preg_match('/^[0-9]{1,2}?.[0-9]{1,2}[a-zA-Z]{1}$/s', $ape);
```
Comment (boherm) [inline classes/Validate.php]: Is the `.` part of your regex is correct?
With this regex, `12.12A` is valid but `12P12C`, `12345A`, `12A` also for instance.

Maybe this regex could be better:
```suggestion
        return (bool) preg_match('/^[0-9]{1,2}?\.?[0-9]{1,2}[a-zA-Z]{1}$/s', $ape);
```
(In test here: https://www.phpliveregex.com/p/K4M)

**Review by boherm — COMMENTED** [review]
Thanks for your contribution @alexandrebak42!
Just a little comment about your regex.

File: classes/Validate.php (hunk 1 above)
Comment (alexandrebak42) [reply classes/Validate.php]: You're right thanks !

**Review by boherm — CHANGES_REQUESTED** [review]
I request changes, just to avoid to merge your PR in his current state.
(following this comment: https://github.com/PrestaShop/PrestaShop/pull/34217#discussion_r1350480073)

**Comment by alexandrebak42** [comment]
Ok sorry I forgot to commit your suggestion

**Review by boherm — DISMISSED** [review]
LGTM now, thanks for your contribution @alexandrebak42 😉

**Comment by alexandrebak42** [comment]
I added a new PR on autoupgrade [here](https://github.com/PrestaShop/autoupgrade/pull/639) to complete this PR.

**Comment by alexandrebak42** [comment]
I close this PR and created a new one on 8.1.x [here](https://github.com/PrestaShop/PrestaShop/pull/34242)

**Pull request closed without merging**